### PR TITLE
Added Chrome Support

### PR DESCRIPTION
**Pull request by TarunKurella**
Hey!
make sure you install the chrome package first
then run the file and open Spotify web

### CODE REVIEW

**Comment by aadibajpai** [comment]
Can you explain what do the additions do? 🙂

**Comment by TarunKurella** [comment]
I've bundled the Chrome Extension in new repository

**Comment by TarunKurella** [comment]
Sure.
I'll detail it later if u don't mind

File: swaglyrics/tab.py [hunk 1]
```diff
@@ -1,21 +1,42 @@
 from swaglyrics.cli import lyrics
-from flask import Flask, render_template
+from flask import Flask, render_template,jsonify,request
+from flask_cors import CORS, cross_origin
 import os
 from swaglyrics import spotify
 
+
 app = Flask(__name__, template_folder=os.path.join(os.path.dirname(os.path.abspath('SwagLyrics-For-Spotify')), 'swaglyrics/templates'))
-# use relative path of the template folder
```
Comment (aadibajpai) [inline swaglyrics/tab.py]: why did you remove this comment?

File: swaglyrics/tab.py [hunk 2]
```diff
@@ -1,21 +1,42 @@
...
-
-@app.route('/')
+@app.route('/',methods=["GET","POST"])
+@cross_origin(supports_credentials=True)
 def tab():
     # format lyrics for the browser tab template
-    global song, artist
-    song = spotify.song()
-    artist = spotify.artist()
+    global song, artist,isChrome,isSameSong
```
Comment (aadibajpai) [inline swaglyrics/tab.py]: spacing is uneven

File: swaglyrics/tab.py [hunk 3]
```diff
@@ -1,21 +1,42 @@
...
+    isSameSong=True
+    if request.method=="POST" or request.method=="OPTIONS" :
+        isChrome=True
+        isSameSong=False
+        response=request.get_json(force=True)
+        song=response["title"]
+        artist=response["artist"]
+        
+
```
Comment (aadibajpai) [inline swaglyrics/tab.py]: why so many newlines?

**Comment by aadibajpai** [comment]
You'll need to add unittests to test the code you have added. Also, what if I want to get the lyrics in the command line interface? 

And if I've got the Spotify desktop app and the web player running at the same time, it will opt for getting the song info via the web player which seems redundant and slower.

What about something like a flag that tells it to get the song and artist info via the chrome extension.
Then it can work with any mode that you choose. Eg. something like `swaglyrics -c --chrome` or `swaglyrics -t -- chrome` where the `--chrome` flag tells it to get the data from the extension.

This is just my opinion so I'd love to know if you have something better in mind.

**Comment by aadibajpai** [comment]
Also, it fetches only when song is changed so if you start Spotify before running the application then it doesn't work until song changes. That should be fixed 👍 

You still need to explain what's happening 🙂

File: README.md [hunk 4]
```diff
@@ -107,6 +107,16 @@ You can quit by pressing <kbd>Ctrl</kbd>+<kbd>C</kbd>
 |-- setup.py
 ```
 
+SwagLyrics Chrome Extension (local testing)
+============================
+ 
+ get it from [here](https://github.com/TarunKurella/SwagLyrics-Chrome).
```
Comment (canihavesomecoffee) [inline README.md]: That link's outdated now :+1:

File: swaglyrics/tab.py (hunk 2 above)
Comment (canihavesomecoffee) [reply swaglyrics/tab.py]: @aadibajpai Add the @pep8speaks integration and this won't happen anymore ;)

Comment (aadibajpai) [reply swaglyrics/tab.py]: @canihavesomecoffee It's there in the repo. I think subsequent changes should work

**Comment by aadibajpai** [comment]
@pep8speaks suggest diff

File: swaglyrics/tab.py [hunk 5]
```diff
@@ -24,9 +45,12 @@ def tab():
...
 def song_changed():
     # to refresh lyrics when song changed
-    global song
-    if song == spotify.song() or spotify.song() is None:
-        return 'no'
+    global song,isChrome,chromeSong
+    if isChrome==False:
+        if song == spotify.song() or spotify.song() is None:
+           return 'no' 
+    if isChrome==True and isSameSong==True:
```
Comment (aadibajpai) [inline swaglyrics/tab.py]: repetitive use of True

File: swaglyrics/tab.py [hunk 6]
```diff
@@ -24,9 +45,12 @@ def tab():
 @app.route('/songChanged', methods=['GET'])
 def song_changed():
     # to refresh lyrics when song changed
-    global song
-    if song == spotify.song() or spotify.song() is None:
-        return 'no'
+    global song,isChrome,chromeSong
+    if isChrome==False:
```
Comment (aadibajpai) [inline swaglyrics/tab.py]:
```suggestion
    if not isChrome:
```

File: swaglyrics/tab.py (hunk 1 above)
Comment (TarunKurella) [reply swaglyrics/tab.py]: I had many different comments as well. So removed all comments for time being

**Comment by TarunKurella** [comment]
> You'll need to add unittests to test the code you have added. Also, what if I want to get the lyrics in the command line interface?
> 
> And if I've got the Spotify desktop app and the web player running at the same time, it will opt for getting the song info via the web player which seems redundant and slower.
> 
> What about something like a flag that tells it to get the song and artist info via the chrome extension.
> Then it can work with any mode that you choose. Eg. something like `swaglyrics -c --chrome` or `swaglyrics -t --chrome` where the `--chrome` flag tells it to get the data from the extension.
> 
> This is just my opinion so I'd love to know if you have something better in mind.

File: swaglyrics/spotify.py [hunk 7]
```diff
@@ -45,20 +45,24 @@ def get_info_linux():
 
 def get_info_mac():
 	from Foundation import NSAppleScript
-	apple_script_code = """
-	getCurrentlyPlayingTrack()
-	on getCurrentlyPlayingTrack()
```
Comment (aadibajpai) [inline swaglyrics/spotify.py]: What do these changes accomplish?

**Comment by aadibajpai** [comment]
@pep8speaks suggest diff

**Comment by TarunKurella** [comment]
Yeah. So SwagLyrics now can be run in 4 modes

- song from local , lyrics in browser
-song from local , lyrics on terminal
- song from browser  , lyrics in  browser
- song from browser , lyrics on terminal

File: swaglyrics/__main__.py [hunk 8]
```diff
@@ -7,66 +7,104 @@
...
-	#  ___) \ V  V / (_| | (_| | |__| |_| | |  | | (__\__ \
-	# |____/ \_/\_/ \__,_|\__, |_____\__, |_|  |_|\___|___/
-	#                     |___/      |___/
-	# 	""")
-	print('Updating unsupported.txt from server.')
-	with open('unsupported.txt', 'w', encoding='utf-8') as f:
-		response = requests.get('http://aadibajpai.pythonanywhere.com/master_unsupported')
-		f.write(response.text)
-	print("Updated unsupported.txt successfully.")
+    # 	print(r"""
```
Comment (aadibajpai) [inline swaglyrics/__main__.py]: Not sure why indentation or quotes changed?

File: swaglyrics/__main__.py [hunk 9]
```diff
@@ -7,66 +7,104 @@
...
-	if args.tab:
-		print('Firing up a browser tab!')
-		app.template_folder = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'templates')
-		app.static_folder = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'static')
-		port = 5042  # random
-		url = "http://127.0.0.1:{port}".format(port=port)
-		threading.Timer(1.25, lambda: webbrowser.open(url)).start()
-		app.run(port=port)
+    # A Function  to initialise Chrome extension Global variables
+    def modeChecker():
```
Comment (aadibajpai) [inline swaglyrics/__main__.py]: Hey, it will be better to use a flag instead of extra prompts so that it is more user friendly.

File: swaglyrics/unsupported.txt [hunk 10]
```diff
@@ -238,3 +238,5 @@ Out Of My Mind - R3HAB Edit by Skytech
 Wherever You Go by Tim Schaufert
```
Comment (aadibajpai) [inline swaglyrics/unsupported.txt]: You can just leave this file empty. The songs will update automatically on first run 👍

**Comment by TarunKurella** [comment]
And for now, application needs to be started before getting songs from Spotify ( extension) due to one-way communication.

I've raised a issue regarding it. I believe it can be solved using web Sockets.

File: swaglyrics/__main__.py (hunk 8 above)
Comment (TarunKurella) [reply swaglyrics/__main__.py]: I actually ran the file through python code formatter
so could be  due to it

File: swaglyrics/__main__.py (hunk 9 above)
Comment (TarunKurella) [reply swaglyrics/__main__.py]: Just felt taking input was more user-friendly imo.

Besides we were already passing an argument already... so multiple parameters just didn't feel good.

However, if you think it is absolutely necessary, I'll change it

File: swaglyrics/spotify.py (hunk 7 above)
Comment (TarunKurella) [reply swaglyrics/spotify.py]: They perform the same action. but the way of getting info is changed

The reason I had to change was because , sometimes, the first code failed to get me the details.

Comment (aadibajpai) [reply swaglyrics/spotify.py]: Did that happen specifically with Chrome or in general as well? Also, you're on a Mac?

File: swaglyrics/__main__.py (hunk 8 above)
Comment (aadibajpai) [reply swaglyrics/__main__.py]: Please revert those :)

File: swaglyrics/__main__.py (hunk 9 above)
Comment (aadibajpai) [reply swaglyrics/__main__.py]: When the person is first running it, they already pass flags. So, passing just one more is more easier than waiting and answering questions. 

Also, it would be easier for a GUI as the GUI could directly run with the params instead of answering the prompts.

File: swaglyrics/__main__.py (hunk 8 above)
Comment (TarunKurella) [reply swaglyrics/__main__.py]: can u mention where specific?

Comment (aadibajpai) [reply swaglyrics/__main__.py]: Whichever you changed. Just make sure the code you add is pep8 compliant 👍

File: swaglyrics/__main__.py [hunk 11]
```diff
@@ -32,10 +32,10 @@ def main():
     )
 
     parser.add_argument(
-        "-t", "--tab", action="store_true", help="Display lyrics in a browser tab."
```
Comment (aadibajpai) [inline swaglyrics/__main__.py]: instead of nargs why don't you add an optional --chrome flag?

File: swaglyrics/__main__.py (hunk 8 above)
Comment (TarunKurella) [reply swaglyrics/__main__.py]: Yupp.
I have used Black Code formatter. and it is pep8 compliant 🙂

File: swaglyrics/spotify.py (hunk 7 above)
Comment (TarunKurella) [reply swaglyrics/spotify.py]: nope. In general for the first time.

yes I am

**Comment by TarunKurella** [comment]
@pep8speaks suggest diff

**Comment by TarunKurella** [comment]
@pep8speaks pep8ify

File: swaglyrics/unsupported.txt (hunk 10 above)
Comment (aadibajpai) [reply swaglyrics/unsupported.txt]: Hasn't been resolved.

File: swaglyrics/__main__.py (hunk 8 above)
Comment (aadibajpai) [reply swaglyrics/__main__.py]: I meant the code you add should be compliant. Don't change the indents of already written code 👍

**Comment by TarunKurella** [comment]
I've Created a new branch called fixing indentations. Please consider that branch.

Also again it failed the test. Turns out as I added chrome args, It just needs to be in the namespace.

So I edited the tests in this( Master)branch. and Yes. it then works.
